cli/src/cmd/generate.rs

```
@@ -0,0 +1,52 @@
+use crate::proto::chisel_rpc_client::ChiselRpcClient;
+use crate::proto::{type_msg::TypeEnum, DescribeRequest};
+use anyhow::{anyhow, Result};
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+pub(crate) async fn cmd_generate(
+    server_url: String,
+    output: PathBuf,
+    version: String,
+) -> Result<()> {
+    let output = File::create(output)?;
+    let mut client = ChiselRpcClient::connect(server_url).await?;
+    let request = tonic::Request::new(DescribeRequest {});
+    let response = execute!(client.describe(request).await);
+    for version_def in response.version_defs {
+        if version_def.version_id != version {
+            continue;
+        }
```

I would do something like:

```
let version_def = response.version_defs.iter().find(|def| dev.version_id == id);
```
if this is `None`, I would return an error saying that there is not such version